e2e/libs/longhorn_deploy/longhorn_deploy.py

```
@@ -18,11 +18,29 @@
+        self.longhorn.create_registry_secret()
+        installed = self.longhorn.install(install_stable_version)
+        if not installed:
+            logging(f"Installing Longhorn failed")
+            time.sleep(self.retry_count)
+            assert False, "Installing Longhorn failed"
+        self.longhorn.setup_longhorn_ui_nodeport()
+
+    def upgrade(self, upgrade_to_transient_version):
+        upgraded = self.longhorn.upgrade(upgrade_to_transient_version)
+        if not upgraded:
+            logging(f"Upgrading Longhorn failed")
```

Ruff (0.8.2)

Undefined name `logging`

(F821)